### PR TITLE
Add i18n logging and messages

### PR DESCRIPTION
**Pull request by ejba**

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/health/HealthLogging.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package io.smallrye.health;
+
+import org.jboss.logging.BasicLogger;
+import org.jboss.logging.Logger;
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+@MessageLogger(projectCode = "SRHCK", length = 5)
```
Comment (xstefank) [inline implementation/src/main/java/io/smallrye/health/HealthLogging.java:10]: are the codes standardized across all SR projects? cc @kenfinnigan, @radcortez

**Comment by xstefank** [comment]
@ejba no, I don't think that tests need to be i18n-ized. I was talking about data messages like https://github.com/smallrye/smallrye-health/blob/master/implementation/src/main/java/io/smallrye/health/checks/InetAddressHealthCheck.java#L49 or https://github.com/smallrye/smallrye-health/blob/master/implementation/src/main/java/io/smallrye/health/checks/UrlHealthCheck.java#L65 or https://github.com/smallrye/smallrye-health/blob/master/implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java#L92.

**Comment by ejba** [comment]
@xstefank I can do it, if you want!

**Comment by xstefank** [comment]
@ejba yes, please do. Better to have it done in one commit :) Thanks!

**Comment by ejba** [comment]
@xstefank Done. Should I add tests to check the logging?

File: implementation/src/main/java/io/smallrye/health/HealthMessages.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.smallrye.health;
+
+import org.jboss.logging.Messages;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+@MessageLogger(projectCode = "SRHCK", length = 5)
+interface HealthMessages {
```
Comment (xstefank) [inline implementation/src/main/java/io/smallrye/health/HealthMessages.java:8]: Why do you need to have this message in separate interface?

File: implementation/src/main/java/io/smallrye/health/checks/HealthChecksLogging.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import org.jboss.logging.BasicLogger;
+import org.jboss.logging.Logger;
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+@MessageLogger(projectCode = "SRHCK", length = 5)
+interface HealthChecksLogging extends BasicLogger {
```
Comment (xstefank) [inline implementation/src/main/java/io/smallrye/health/checks/HealthChecksLogging.java:11]: I am not that much familiar with JBoss logging, but can't we have only single logger class with all messages? Or is that an antipattern?

File: implementation/src/main/java/io/smallrye/health/HealthLogging.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+interface HealthLogging extends BasicLogger {
+    HealthLogging log = Logger.getMessageLogger(HealthLogging.class, HealthLogging.class.getPackage().getName());
+
+    @LogMessage(level = Logger.Level.ERROR)
+    @Message(id = 1000, value = "Error processing Health Checks")
+    void healthCheckError(@Cause Throwable throwable);
+
+    @LogMessage(level = Logger.Level.ERROR)
```
Comment (xstefank) [inline implementation/src/main/java/io/smallrye/health/HealthLogging.java:18]: This is info in the current code.

**Comment by xstefank** [comment]
@ejba I don't think that is necessary.

File: implementation/src/main/java/io/smallrye/health/checks/HealthChecksLogging.java (hunk 3 above)
Comment (ejba) [reply implementation/src/main/java/io/smallrye/health/checks/HealthChecksLogging.java:11]: Me neither. Just follow the procedure [here](https://github.com/smallrye/smallrye-parent/wiki/I18N-Messages-and-Logging#general-requirements). 

If you feel that is something should be changed, tell me and I will do it.

File: implementation/src/main/java/io/smallrye/health/HealthLogging.java (hunk 4 above)
Comment (ejba) [reply implementation/src/main/java/io/smallrye/health/HealthLogging.java]: whoops, sorry! Fixed.

File: implementation/src/main/java/io/smallrye/health/HealthMessages.java (hunk 2 above)
Comment (ejba) [reply implementation/src/main/java/io/smallrye/health/HealthMessages.java:8]: I assumed that I have to, according with this [documentation](https://github.com/smallrye/smallrye-parent/wiki/I18N-Messages-and-Logging), but I can be wrong.

Comment (xstefank) [reply implementation/src/main/java/io/smallrye/health/HealthMessages.java:8]: @ejba thank you for the link, I must have missed that the logging was standardized :)

**Comment by ejba** [comment]
@xstefank I am sorry, I forgot to do that. Glad to contribute 😄

**Pull request merged**